### PR TITLE
docs: Clefairy release notes

### PR DESCRIPTION
**Pull request by markcaron**
## What I did

1. Added version 2.1.0 release notes
2. Used the `compact` tag size for the tables

## Testing Instructions

1. Check accuracy of the release notes table

### CODE REVIEW

File: docs/release-notes/index.md [hunk 1]
```diff
@@ -66,6 +66,46 @@ or anything else, please submit a [GitHub issue][issues].
...
+## Version 2.1.0
+Released October 1, 2024
+
+### Highlights
+
+<rh-table>
+
+| Change                              | Type    | Notes |
+| ----------------------------------- | ------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Theming: added theming tokens to most elements | {{i()}} | New theming tokens allow page-level and container-level colour palettes on patterns and elements. |
```
Comment (hellogreg) [inline docs/release-notes/index.md:80]: This is the only entry that ends with a period--in v2.1 anyway. But all of the <=v2.0 entries end in periods.

Comment (markcaron) [reply docs/release-notes/index.md:80]: This is why I asked you to review 😅

File: docs/release-notes/index.md [hunk 2]
```diff
@@ -66,6 +66,46 @@ or anything else, please submit a [GitHub issue][issues].
...
+| `<rh-skip-link>`: added optional `href` attribute | {{i()}} | Adding `href` attribute to `<rh-skip-link>` removes the need for slotting an anchor element (`<a>`). |
+| `<rh-audio-player>`: added mini playback control | {{p()}} | Mini layout now has playback control. |
+| `<rh-navigation-secondary>`: improved accessibility | {{p()}} | Removed arrow-key keyboard navigation in favor of tab navigation through navigation items and added `accessible-label` attribute to explicitly label landmark. |
+| `<rh-subnav>`: improved accessibility | {{p()}} | Removed arrow-key keyboard navigation in favor of tab navigation through navigation items and added `accessible-label` attribute to explicitly label landmark. |
+| `<rh-table>`: added container query support | {{p()}} | Tables now adjust to the size of their containing element, not just viewport size. |
+| `<rh-tabs>`: added container query support | {{p()}} | Tabs now adjust to the size of their containing element, not just viewport size. |
```
Comment (zeroedin) [inline docs/release-notes/index.md:98]:
```
| `<rh-tabs>`: added container query support | {{p()}} | Tabs now adjust to the size of their containing element, not the viewport size. |
```

File: docs/release-notes/index.md [hunk 3]
```diff
@@ -66,6 +66,46 @@ or anything else, please submit a [GitHub issue][issues].
...
+| `<rh-tile>`: added `private` and `external` link variants | {{i()}} |  Using `link="private"` or `link="external"` indicates whether the link is private or external and changes the tile icon from an arrow to a padlock or external link icons respectively. |
+| `<rh-skip-link>`: added optional `href` attribute | {{i()}} | Adding `href` attribute to `<rh-skip-link>` removes the need for slotting an anchor element (`<a>`). |
+| `<rh-audio-player>`: added mini playback control | {{p()}} | Mini layout now has playback control. |
+| `<rh-navigation-secondary>`: improved accessibility | {{p()}} | Removed arrow-key keyboard navigation in favor of tab navigation through navigation items and added `accessible-label` attribute to explicitly label landmark. |
+| `<rh-subnav>`: improved accessibility | {{p()}} | Removed arrow-key keyboard navigation in favor of tab navigation through navigation items and added `accessible-label` attribute to explicitly label landmark. |
+| `<rh-table>`: added container query support | {{p()}} | Tables now adjust to the size of their containing element, not just viewport size. |
```
Comment (zeroedin) [inline docs/release-notes/index.md:97]:
```Suggestion
| `<rh-table>`: added container query support | {{p()}} | Tables now adjust to the size of their containing element, not the viewport size. |
```

File: docs/release-notes/index.md (hunk 2 above)
Comment (zeroedin) [reply docs/release-notes/index.md:98]: Media queries were removed completely, the "added support" here was more or less a replacement of @media for @container

File: docs/release-notes/index.md (hunk 3 above)
Comment (zeroedin) [reply docs/release-notes/index.md:97]: Media queries were removed completely, the "added support" here was more or less a replacement of @media for @container

File: docs/release-notes/index.md (hunk 2 above)
Comment (markcaron) [reply docs/release-notes/index.md:98]: Was basing it off the [changeset](https://github.com/RedHat-UX/red-hat-design-system/blob/34b87ff1cd73bd28659348ce6400efb47ea78555/.changeset/large-geese-suffer.md). Maybe I had the wording wrong?

**Review by hellogreg — APPROVED** [review]
la grammaire: très magnifique

**Pull request merged**